src/collector.ts

```diff
@@ -124,22 +127,35 @@ class NaiveGomodParser {
                         const entry: IKeyValueEntry = new KeyValueEntry(pkgName, { line: 0, column: 0 });
                         entry.value = new Variant(ValueType.String, 'v' + version[0]);
                         entry.value_position = { line: index + 1, column: version.index };
-                        dependencies.push(new Dependency(entry));
-
-                        // Find packages of this module.
-                        goImports.forEach(pckg => {
-                            if (pckg != pkgName && pckg.startsWith(pkgName + "/")) {
-                                const entry: IKeyValueEntry = new KeyValueEntry(pckg, { line: 0, column: 0 });
-                                entry.value = new Variant(ValueType.String, 'v' + version[0]);
-                                entry.value_position = { line: index + 1, column: version.index };
-                                dependencies.push(new Dependency(entry));
-                            }
-                        })
+                        dependencies.push(new Dependency(entry, pkgName));
                     }
                 }
             }
             return dependencies;
         }, []);
+
+        const totalDirectDeps = dependencies.length
+        goImports.forEach(pckg => {
+            let exactMatchDep: Dependency = null;
+            let moduleMatchDep: Dependency = null;
+            for (let i = 0; i < totalDirectDeps; i++) {
+                let dep: Dependency = dependencies[i];
+                if (pckg == dep.name.value) {
+                    exactMatchDep = dep;
+                } else if (pckg.startsWith(dep.name.value + "/")) {
+                    moduleMatchDep = dep;
+                }
```

Comment:
This logic may not work for the following test,

Imports:
```
github.com/googleapis/gax-go/abc
fmt
github.com/googleapis/gax-go/v2
```

go.mod
```go.mod
      module test/data/sample1

      go 1.15

      require (
        github.com/googleapis/gax-go v1.0.3
        github.com/googleapis/gax-go/v2 v2.0.5
      )

```

Reply:
Current logic works for this input, I have added one more test case as below: